File: 2021/April/Day 29/truncate_sentence.py
# Source : https://leetcode.com/problems/truncate-sentence/submissions/
class Solution:
    def truncateSentence(self, s: str, k: int) -> str:
        l = s.split(' ');

        new_s = ""
        for i in range(k):
            new_s = new_s + l[i]
            if(i!=k-1):
                new_s = new_s + " "
        return new_s

File: 2023/March/Day 6/truncateSentence.cpp
// Source : https://leetcode.com/problems/truncate-sentence/submissions/
class Solution {
public:
    string truncateSentence(string s, int k) {
        
        string out ="";
        
        int spaces=0;
        
        for(auto&i:s)
        {
            if(i==' ')
            {
                spaces++;
            }
            if(spaces<k)
            {
                out+=i;
            }
            
        }
        
        return out;
        
    }
};